src/pool.spec.ts

```diff
@@ -401,6 +402,51 @@ describe("Constant product pool", () => {
     await pool.updateState();
     expect(pool.state.totalLiquidity).toBe(1000);
   });
+
+  it("zap e2e scenario", async () => {
+    const { account, pool, coin } = await makeFreshTestBed({
+      poolType: "CONSTANT_PRODUCT",
+    });
+
+    expect(pool.state).toEqual({
+      totalLiquidity: 0,
+      totalPrimary: 0,
+      totalSecondary: 0,
+      primaryAssetPrice: 0,
+      secondaryAssetPrice: 0,
+    });
+
+    // Opt in for liquidity asset.
+    const liqOptInTx = await pool.liquidityAsset.prepareOptInTx(account.addr);
+    await signAndSend(liqOptInTx, account);
+
+    // Add first liquidity.
+    await addLiquidity(account, pool, 100_000, 100_000);
+
+    // Zapping.
+    const zap = pool.prepareZap({
+      asset: coin,
+      amount: 10_000,
+      slippagePct: 1,
+    });
+    const zapParams = zap.params;
+
+    expect(zapParams.swapDeposited + zapParams.secondaryAddLiq).toBe(10_000n);
+    expect(zapParams.swapReceived >= zapParams.primaryAddLiq).toBe(true);
```

Comment:
generally thing we get from swap should be passed to add liquidity, I think here might be a problem with returning `swapReceived` in zap research script I wrote